cairo_vm_hints/src/hints/lib/verifiers/evm/block_tx_verifier.rs

```diff
@@ -0,0 +1,159 @@
+use cairo_vm::{
+    hint_processor::builtin_hint_processor::{
+        builtin_hint_processor_definition::HintProcessorData,
+        hint_utils::{get_integer_from_var_name, get_ptr_from_var_name, insert_value_from_var_name, insert_value_into_ap},
+    },
+    types::exec_scope::ExecutionScopes,
+    vm::{errors::hint_errors::HintError, vm_core::VirtualMachine},
+    Felt252,
+};
+use std::collections::HashMap;
+
+use crate::{
+    hint_processor::models::proofs::{transaction::Transaction, Proofs},
+    hints::{lib::utils::count_leading_zero_nibbles_from_hex, vars},
+};
+
+pub const HINT_BATCH_TRANSACTIONS_LEN: &str = "memory[ap] = to_felt_or_relocatable(len(batch.transactions))";
+
+pub fn hint_batch_transactions_len(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    _hint_data: &HintProcessorData,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let batch = exec_scopes.get::<Proofs>("batch")?;
+
+    insert_value_into_ap(vm, Felt252::from(batch.transactions.len()))
+}
+
+pub const HINT_SET_TX: &str = "transaction = transactions[ids.idx]";
+
+pub fn hint_set_tx(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    hint_data: &HintProcessorData,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let batch = exec_scopes.get::<Proofs>("batch")?;
+    let idx: usize = get_integer_from_var_name(vars::ids::IDX, vm, &hint_data.ids_data, &hint_data.ap_tracking)?
+        .try_into()
+        .unwrap();
+    let transaction = batch.transactions[idx].clone();
+
+    exec_scopes.insert_value::<Transaction>("transaction", transaction);
+
+    Ok(())
+}
+
+pub const HINT_SET_TX_KEY: &str = "from tools.py.utils import split_128\n(ids.key.low, ids.key.high) = split_128(int(transaction.key, 16))";
+
+pub fn hint_set_tx_key(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    hint_data: &HintProcessorData,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let transaction = exec_scopes.get::<Transaction>("transaction")?;
+    let key = transaction.key;
+
+    let key_as_limbs = key.as_limbs();
+    let key_low = key_as_limbs[0] as u128 | ((key_as_limbs[1] as u128) << 64);
```

Comment:
u128 is going to over flow in account u did it right why not do it same way here?